pkg/clustermanager/cluster.go

```diff
@@ -152,10 +160,13 @@ func (manager *Manager) SetupEncryptedNetwork() error {
 func (manager *Manager) InstallMasters() error {
 
 	commands := []NodeCommand{
-		{"kubeadm init", "kubeadm init --config /root/master-config.yaml"},
+		{"kubeadm init", "kubectl version > /dev/null &> /dev/null || kubeadm init --config /root/master-config.yaml"},
 		{"configure kubectl", "rm -rf $HOME/.kube && mkdir -p $HOME/.kube && cp -i /etc/kubernetes/admin.conf $HOME/.kube/config && chown $(id -u):$(id -g) $HOME/.kube/config"},
-		{"install flannel", "kubectl apply -f https://raw.githubusercontent.com/coreos/flannel/v0.10.0/Documentation/kube-flannel.yml"},
-		{"configure flannel", "kubectl -n kube-system patch ds kube-flannel-ds --type json -p '[{\"op\":\"add\",\"path\":\"/spec/template/spec/tolerations/-\",\"value\":{\"key\":\"node.cloudprovider.kubernetes.io/uninitialized\",\"value\":\"true\",\"effect\":\"NoSchedule\"}}]'"},
+		{"install Weave Net", "kubectl apply -f \"https://cloud.weave.works/k8s/net?k8s-version=$(kubectl version | base64 | tr -d '\\n')\""},
```

Comment:
I cannot confirm for sure if there was a problem with canal because I switched to Weave Net following few articles which were recommending it for its easy no config setup. I am not sure what is the advantage of canal though..

Reply:
it allows to enforce Pod Security Policies as of using callico. It is actually the same simple to setup as Weave or Flannel....

Reply:
I just tested it and it works fine. Even without the ip routing command. I am not sure which we should stick to.

Reply:
I would prefer canal over Weave, if that works

Reply:
I switched back to it. Although I see weave is much more popular..